x-pack/plugins/security_solution/common/endpoint/models/event.ts

```diff
@@ -52,3 +52,123 @@ export function parentEntityId(event: ResolverEvent): string | undefined {
   }
   return event.process.parent?.entity_id;
 }
+
+/**
+ * @param event The event to get the category for
+ */
+export function eventCategory(event: ResolverEvent): string {
+  // Returning "Process" as a catch-all here because it seems pretty general
+  let eventCategoryToReturn: string = 'Process';
+  if (isLegacyEvent(event)) {
+    const legacyFullType = event.endgame.event_type_full;
+    if (legacyFullType) {
+      return legacyFullType;
+    }
+  } else {
+    const eventCategories = event.event.category;
+    const category =
+      typeof eventCategories === 'string' ? eventCategories : eventCategories[0] || '';
+
+    if (category) {
+      eventCategoryToReturn = category;
+    }
+  }
+  return eventCategoryToReturn;
+}
+
+/**
+ * ECS event type will be things like 'creation', 'deletion', 'access', etc.
+ * see: https://www.elastic.co/guide/en/ecs/current/ecs-event.html
+ * @param event The ResolverEvent to get the ecs type for
+ */
+export function ecsEventType(event: ResolverEvent): string {
+  if (isLegacyEvent(event)) {
+    return event.endgame.event_subtype_full || '';
+  }
+  return typeof event.event.type === 'string' ? event.event.type : event.event.type.join('/');
+}
+
+/**
+ * #Descriptive Names For Related Events:
+ *
+ * The following section provides facilities for deriving **Descriptive Names** for ECS-compliant event data.
+ * There are drawbacks to trying to do this: It *will* require ongoing maintenance. It presents temptations to overarticulate.
+ * On balance, however, it seems that the benefit of giving the user some form of information they can recognize & scan outweighs the drawbacks.
+ */
+
+/**
+ * A type that annotates the basic requirements for giving the event a `descriptive name`
+ */
+type detailRecord<T> = T extends 'registry'
+  ? { path: unknown; key: unknown }
+  : T extends 'dns'
+  ? { question: { name: unknown } }
+  : T extends 'network'
+  ? { direction: unknown; forwarded_ip: unknown }
+  : T extends 'file'
+  ? { path: unknown }
+  : unknown;
+type ecsRecordWithType<T extends string> = Record<T, detailRecord<T>>;
+
+/**
+ * Verify that the `ecsCategory` exists as a key on the event record. i.e. if ecsCategory is `registry`, the event is shaped like {registry: object}
+ *
+ * @param event
+ * @param ecsCategory
+ */
+export function isRecordNamable<T extends string>(
+  event: object,
+  ecsCategory: T
+): event is ecsRecordWithType<T> {
+  // The goal here is to reach in and grab a better name for the event (if one exists) without being too obtrusive/assertive about ECS
+  return typeof (event as ecsRecordWithType<T>)[ecsCategory] === 'object';
```

Comment:
Well that sucks, because I was planning on asking you for help with this. Maybe I'll explain what's happening here better and see if we can come up with a more terse solution.

Reply:
per our discussion: let's pair on this.